cmd/html/script.js

```diff
@@ -0,0 +1,200 @@
+document.addEventListener("DOMContentLoaded", () => {
+
+    // create nodes on svg based on getNodes() function
+    mapPodsToNodes().then(nodeData=>{
+        // create svg
+        const svg = d3.select("#visualization")
+        .append("svg")
+        .attr("width", window.innerWidth)
+        .attr("height", window.innerHeight);
+        
+        console.log(nodeData)
+        // create a simple root node
+        const root = d3.hierarchy(nodeData)
+        .sum(d => computeSize(d))
+        .sort();
+        
+        // print root node
+    console.log("Root node",root)
+
+    // create a treemap with nodeData
+    const treemap = d3.treemap()
+    // .title("kubernetes time machine")
+    // window.innerWidth, window.innerHeight
+    .size([600,600])
+        .paddingInner(10)
+        .paddingTop(2)
+        .paddingRight(2)
+        .paddingBottom(2)
+        .paddingLeft(2)
+        .round(2)
+        // .padding(0)
+        .tile(d3.treemapSquarify.ratio(1))(root)
+
+        console.log("Treemap",treemap)
+        // console.log(treemap.leaves())    
+        
+    const g = svg.append("g")
+    .attr("transform", "translate(" + treemap.leaves()[0].x0 + "," + treemap.leaves()[0].y0 + ")");
+    
+
+    
+
+    // add nodes
+    g.selectAll("a")
+    .data(treemap.descendants().slice(0))
+    .enter()
+    .append("a")
+    .attr("href", (d) => {
+      if (d.depth === 0) {
+        return "#"
+      }
+      return d.height === 1 ? "node/"+d.data.name : "pod/"+d.data.name
+      return d.data.name+"/"
+    
+    })
+
+    // add rectangles for each pod
+    g.selectAll("a")
+    .data(treemap.descendants().slice(0))
+    .append("rect")
+    .attr("x", (d) => {
+        // console.log(d.x0, d.x1);
+        // console.log(d)
+        return d.x0-0;
+    })
+    .attr("y", (d) => {
+        // console.log(d.y0, d.y1);
+        return d.y0-0;
+    })
+    .attr("width", (d) => (d.x1 - d.x0)+0)
+    .attr("height", (d) => (d.y1 - d.y0)+0)
+    .style("fill", (d) => {
+        // console.log("DATA type:", d);
+        if (d.depth === 0) {
+            return "lightblue"
+        }
+        return d.height === 1 ? "#03385c" : "#3568a3"
+    })
+    .attr("margin", (d) => {
+        if (d.depth === 0) {
+            return "0px"
+        }
+        return d.height === 1 ? "5px" : "10px"
+    });
+    
+    // add text for each pod
+    // g.selectAll("a")
+    // .data(treemap.descendants().slice(0))
+    // // .enter()
+    // .append("text")
+    // .attr("x", (d) => d.x0 + 10)
+    // .attr("y", (d) => {
+    //     if (d.depth === 0) {
```

Comment:
When I have some time I'll try and make a branch with a framework and some Jest tests for certain things and react-testing-library for others.

it'll make it much less risky to experiment and cause regressions. It'll also make it more _fun_

Reply:
Thanks! That'd be great. Currently, I'm using using this frontend code as a way to validate that my server is returning proper responses.